pkg/kubectl/cmd/exec.go

```diff
@@ -254,7 +261,8 @@ func (p *ExecOptions) Run() error {
 
 	containerName := p.ContainerName
 	if len(containerName) == 0 {
-		glog.V(4).Infof("defaulting container name to %s", pod.Spec.Containers[0].Name)
+		msg := fmt.Errorf("defaulting container name to %s, use '%s describe po/%s' cmd to see all containers in this pod", pod.Spec.Containers[0].Name, p.FullCmdName, p.PodName)
```

Comment:
Detect whether there's only one container.  If there's only one container, there's no reason to drop this message.